src/Rings/AbelianClosure.jl

```diff
@@ -526,6 +526,41 @@ end
 (R::QQField)(a::QQAbFieldElem) = R(a.data)
 (R::ZZRing)(a::QQAbFieldElem) = R(a.data)
 
+################################################################################
+#
+#  Conversion to `QQBarFieldElem`
+#
+#  We assume the natural embedding of cyclotomic fields into `QQBarField()`
+#  that is given by the (documented) fact that
+#  `root_of_unity(F::QQBarField, n::Int)` is $\exp(2 \pi i / n)$.
+#
+################################################################################
+
+function (F::QQBarField)(a::QQAbFieldElem)
+  N = a.c
+  cfs = Oscar.coefficients(a.data)
+  r = root_of_unity(F, N)
+  pow = one(F)
+  res = cfs[1] * pow
+  for i in 2:length(cfs)
+    pow = pow * r
+    res = res + cfs[i] * pow
+  end
+  return res
+end
+
+Nemo.QQBarFieldElem(a::QQAbFieldElem) = algebraic_closure(QQ)(a)
+
+################################################################################
+#
+#  Conversion to `Float64`, `ComplexF64`
+#
+#  We first convert to a `QQBarFieldElem` and then use that it supports the
+#  conversions in question.
+#
+
+Core.Float64(a::QQAbFieldElem) = Float64(algebraic_closure(QQ)(a))
+Base.ComplexF64(a::QQAbFieldElem) = ComplexF64(algebraic_closure(QQ)(a))
```

Comment:
Since you added the `QQBarFieldElem` constructor above, might as well do this? But I am also happy to leave it as-is
```suggestion
Core.Float64(a::QQAbFieldElem) = Float64(QQBarFieldElem(a))
Base.ComplexF64(a::QQAbFieldElem) = ComplexF64(QQBarFieldElem(a))
```